zcash_client_backend/src/encoding.rs

```diff
@@ -36,6 +38,61 @@ where
     }
 }
 
+pub trait AddressCodec<P>
```

Comment:
I agree, let's not add public APIs that we intend to remove.

Reply:
This doesn't quite have the same semantics as `zcash_address` provides: this does the full decoding to domain types, whereas zcash_address only provides decoding to bytes. I think that this is generally a useful trait to have; I'm not sure if it could be made crate-private given that I think it's likely used in public APIs.

Reply:
We will need decoding to domain types somewhere, but this is probably the wrong place.

Reply:
So, I tried making this crate-private and that doesn't work, because it's required in zcash_client_sqlite. So where should it go?

Reply:
Unfortunately there's no way to mark code outside of the standard library as unstable in rustdoc. https://internals.rust-lang.org/t/stability-attributes-for-third-party-libraries/1631